go/arrow/flight/flightsql/driver/utils.go

```diff
@@ -60,44 +60,135 @@ func (g grpcCredentials) RequireTransportSecurity() bool {
 func fromArrowType(arr arrow.Array, idx int) (interface{}, error) {
 	switch c := arr.(type) {
 	case *array.Boolean:
+		if c.IsNull(idx) {
+			return nil, nil
+		}
+
 		return c.Value(idx), nil
 	case *array.Float16:
+		if c.IsNull(idx) {
+			return nil, nil
+		}
+
 		return float64(c.Value(idx).Float32()), nil
 	case *array.Float32:
+		if c.IsNull(idx) {
+			return nil, nil
+		}
+
 		return float64(c.Value(idx)), nil
 	case *array.Float64:
+		if c.IsNull(idx) {
+			return nil, nil
+		}
+
 		return c.Value(idx), nil
+	case *array.Decimal128:
+		if c.IsNull(idx) {
+			return nil, nil
+		}
+
+		dt, ok := arr.DataType().(*arrow.Decimal128Type)
+		if !ok {
+			return nil, fmt.Errorf("datatype %T not matching decimal128", arr.DataType())
+		}
+
+		return c.Value(idx).ToFloat64(dt.Scale), nil
+	case *array.Decimal256:
+		if c.IsNull(idx) {
+			return nil, nil
+		}
+
+		dt, ok := arr.DataType().(*arrow.Decimal256Type)
+		if !ok {
+			return nil, fmt.Errorf("datatype %T not matching decimal256", arr.DataType())
+		}
+
+		return c.Value(idx).ToFloat64(dt.Scale), nil
 	case *array.Int8:
+		if c.IsNull(idx) {
+			return nil, nil
+		}
+
 		return int64(c.Value(idx)), nil
```

Comment:
i know that this is the existing code, but since we're in this discussion, does it make sense to upcast to int64 rather than letting it stay int8?